# Chapter 6 입력과 출력
# 6.1 화면 출력
## print 

In [1]:
print("Hello Python!!")

Hello Python!!


In [2]:
print("Best", "python", "book")

Best python book


In [3]:
print("Best", "python", "book", sep = "-:*:-")

Best-:*:-python-:*:-book


In [4]:
print("abcd" + "efg")

abcdefg


In [5]:
print("Best", "python", "book" + ":", "This book")

Best python book: This book


In [6]:
name = "James"
ID_num = 789
print("Name:", name + ",", "ID Number:", ID_num )

Name: James, ID Number: 789


`\n` 줄바꿈 표시 (개행문자)

In [8]:
print("James is my friend.\n \n He is Korean.")

James is my friend.
 
 He is Korean.


다른 문자열 추가 방법 (enter 안치고 줄 바꿈없이)
`end = 문자열`

In [10]:
print("Welcome to ", end="")
print("python!")

Welcome to python!


In [13]:
print("Welcome to ", end="programming ")
print("python!")

Welcome to programming python!


나머지 연산자(%) 이용

`print("%type" % data)`
* %type 에서 문자열이면 %s(string), 정수면 %d(혹은 %i), 실수면 %f(%F도 가능). 실수로 %f은 6자리까지 출력

data가 두 개 이상이면   
`print("%type %type" % (data1,data2))`

문자열을 대입한 변수를 출력(%s 이용)

In [14]:
name = "광재"
print("%s는 나의 친구입니다." % name)

광재는 나의 친구입니다.


%d와 %f를 이용해 정수와 실수를 출력

In [15]:
r = 3  # 변수 r에 정수 데이터 할당
PI = 3.14159265358979 # 변수 PI에 실수 데이터 할당
print("반지름: %d, 원주율: %f" % (r, PI)) # 지정된 위치에 데이터 출력

반지름: 3, 원주율: 3.141593


### 형식 지정 문자열에서 [출력 위치] 지정

`print()` 함수에서 `string.format()`을 이용해 형식 지정 문자열 기본 구조 파악하기   
`print("{0},{1}, ... {n})".format(data_0,data_1,data_2, ..., data_n)`

In [16]:
animal_0 = "cat"
animal_1 = "dog"
animal_2 = "fox"

print("Animal: {0}".format(animal_0))
print("Animal: {0},{1},{2}".format(animal_0, animal_1, animal_2))

Animal: cat
Animal: cat,dog,fox


In [17]:
print("Animal: {1},{2},{0}".format(animal_0, animal_1, animal_2))

Animal: dog,fox,cat


In [18]:
print("Animal: {0},{2}".format(animal_0, animal_1, animal_2))

Animal: cat,fox


`format()`안의 데이터를 순차적으로 지정할 때 빈 중괄호여도 됨.

In [19]:
print("Animal: {}, {}, {}".format(animal_0, animal_1, animal_2))

Animal: cat, dog, fox


In [20]:
name = "Tomas"
age = 10
a = 0.1234567890123456789
fmt_string = "String: {0}. Integer Number: {1}. Floating Number: {2}"
print(fmt_string.format(name, age, a))

String: Tomas. Integer Number: 10. Floating Number: 0.12345678901234568


### 형식 지정 문자열에서 숫자 출력 형식 지정

In [21]:
a = 0.1234567890123456789
print("{0:.2f}, {0:.5f}".format(a))

0.12, 0.12346


<표> 숫자의 출력 형식 지정   

데이터(x) | 출력 형식 | 출력 결과 | 설명  
 --- | --- | --- | ---   
 3 | `N:2d` | ↔3 | 정수를 공백 포함해 두 자리료 표시(공백 한 칸)   
 3  | `N:05d` | 00003 |  정수를 다섯 자리로 표시. 앞의 공백은 0으로 채움   
 12  | `N:>5d` | ↔↔↔12 |  정수를 다섯 자리로 표시. 숫자는 오른촉으로 정렬   
 0.12345  | `N:.3f` | 0.123 |  실수를 소숫점 셋째 자리까지 표시   
 745600  | `N:,` | 7,456,000 |  통화 표시처럼 끝에서 셋째 자리마다 반점(,) 표시   
 0.3258  | `N:1%` | 32.6% |  소수를 %로 표시. 퍼센트 표시에서 소수점 자리 수는 '.' 다음 숫자로 표시   
 92500000000  | `N:.2e` | 9.25e+10 |  숫자를 지수로 표시. 지수 표시에서 소수점 자리 수는 '.' 다음 숫자로 표시   
 16  | `N:#x` | 0x10 | 숫자를 16진수로 표시. #기호가 없으면 0x 없이 출력됨.   
 8  | `N:#o` | 0o10 |  숫자를 8진수로 표시. #기호가 없으면 0o 없이 출력됨.    
 2  | `N:#b` | 0b10 |  숫자를 2진수로 표시. #기호가 없으면 0b 없이 출력됨. 



# 6.2 입력
`input()` - 키보드로 데이터 입력.   
`data=input("문자열")`